assets/src/hooks/use-event-date-update-related-tickets.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * External imports.
+ */
+import { useCallback } from '@wordpress/element';
+import { parseInfinity } from '@eventespresso/utils';
+/**
+ * Internal imports
+ */
+import useEventDateTickets from './use-event-date-tickets';
+
+/**
+ * A custom react hook to update the related ticket entities for the given
+ * datetime entity from the eventespresso/core store state.
+ *
+ * @param {BaseEntity} eventDate  A datetime BaseEntity instance.
+ */
+const useEventDateUpdateRelatedTickets = ( eventDate ) => {
+	const { tickets: relatedTickets, ticketsLoaded } = useEventDateTickets( eventDate );
+	return useCallback( ( { capacity = null } ) => {
+		if ( ticketsLoaded ) {
+			// Make sure that the ticket qty value is compared with
+			// a non negative datetime capacity value in Math.min()
+			const nonNegativeDTTCapacity = parseInfinity( capacity, false, false );
+			relatedTickets.forEach( ( ticket ) => {
+				if ( capacity !== null ) {
```

Comment:
since we don't need to do anything when capacity is null, then why not merge this with the conditional above and then bypass everything?

ie:

```js
if ( ticketsLoaded && capacity !== null ) {
    // loop through tickets and possibly adjust quantity
}
```

Reply:
`parseInfinity` already converts `null` to `Infinity` or `-1` by making use of `isNil( value );`

Reply:
My comment has nothing to do with `parseInfinity` or how it converts values.

Look at the entire function again:


```js
	( { capacity = null } ) => {
		if ( ticketsLoaded ) {
			// Make sure that the ticket qty value is compared with
			// a non negative datetime capacity value in Math.min()
			const nonNegativeDTTCapacity = parseInfinity( capacity, false, false );
			relatedTickets.forEach( ( ticket ) => {
				if ( capacity !== null ) { <<<--- WE ARE CHECKING THIS REPEATEDLY
					const nonNegativeTKTQty = parseInfinity( ticket.qty, false, false );
					const qty = Math.min( nonNegativeDTTCapacity, nonNegativeTKTQty );
					ticket.qty = parseInfinity( qty, true, true );
				}
			} );
		}
	}
```

`capacity` is passed into the callback but is not modified in any way before being compared against null while looping through tickets to decide whether to adjust the ticket qty. But since the value of `capacity` is never changed anywhere in the function, and we don't end up doing anything when it's incoming value is `null`, we are running that same comparison over and over. We could move that comparison into the very first conditional and then bypass **everything** when `capacity` is `null` since we are not going to end up doing anything anyways.

```js
if ( ticketsLoaded && capacity !== null ) { <<<--- ONLY NEED TO CHECK CAP ONCE
	// Make sure that the ticket qty value is compared with
	// a non negative datetime capacity value in Math.min()
	const nonNegativeDTTCapacity = parseInfinity( capacity, false, false );
	relatedTickets.forEach( ( ticket ) => {
		const nonNegativeTKTQty = parseInfinity( ticket.qty, false, false );
		const qty = Math.min( nonNegativeDTTCapacity, nonNegativeTKTQty );
		ticket.qty = parseInfinity( qty, true, true );
	} );
}
```

Reply:
I was pointing towards how the value actually reaches there. If you check the `onChange` handler for `capacity`,
```JS
onChange: ( cap ) => {
	cap = parseInfinity( cap, true, true );
	eventDate.regLimit = cap;
	updateRelatedTickets( { capacity: cap } );
	return <InfinitySymbol value={ cap } asInt />;
}
```
You can see that the value is not null.
May be we can pass the actual value of `cap` from the `onChange` handler to the hook for updating the ticket quantity.

Reply:
Why I have used `if ( capacity !== null )`?
As I mentioned in the commit message, it is a generic hook which can/may be used to update other fields in the related tickets. This is also the reason that the arguments are passed as object `( { capacity = null } )` instead of `( capacity = null )`.

Reply:
> 
> 
> Why I have used `if ( capacity !== null )`?
> As I mentioned in the commit message, it is a generic hook which can/may be used to update other fields in the related tickets. This is also the reason that the arguments are passed as object `( { capacity = null } )` instead of `( capacity = null )`.

@tn3rb I understand your point, please read this quoted comment again. What if in future we want to update something other than quantity for the related tickets? We could use the same hook and that time capacity may not need to be changed. I hope I am not trying create an ultra hypothetical use case. 😄

Reply:
**GO FOR IT!!!**  😆

Reply:
Done! Please have a look 😊